core/src/main/java/org/apache/spark/unsafe/map/BytesToBytesMap.java

```diff
@@ -741,14 +741,6 @@ public boolean append(Object kbase, long koff, int klen, Object vbase, long voff
         numKeys++;
         longArray.set(pos * 2 + 1, keyHashcode);
         isDefined = true;
-
-        if (numKeys > growthThreshold && longArray.size() < MAX_CAPACITY) {
```

Comment:
longArray.size() is the next capacity for current grow strategy, it should be `longArray.size() <= MAX_CAPACITY`